### ISSUE TITLE
Run 32bit test suite on PR CI?

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#69823]
@rust-lang/infra do we have enough spare capacity on the PR CI runner to add something like `./x.py test src/test/ui --target i686-unknown-linux-gnu --pass check`? That would help find 32bit-only issues in tests before things get to bors. And running with `--pass check` would also ensure that the test suite actually passes in check-only mode, which is not a given (some tests fail and need `ignore-pass`).

**Comment by pietroalbini** [comment]
Nominating for discussion at the infrastructure team meeting.

**Comment by Mark-Simulacrum** [comment]
I would be interested in seeing error rates, at least; I have not experienced a lot of pain with 32 bit tests failing (and 64 bit tests passing). OTOH, maybe the `--pass check` mode is worth it...

Do we have numbers on how long we expect this to take on CI? I would expect someone to run the UI tests that we already run on CI locally and then figure out an approximate "CI slowdown" based on how long those take, to determine how long we'd expect this test to take.

**Comment by pietroalbini** [comment]
We discussed this in the infra team meeting, and we'd like to see numbers on the impact this would have in our CI times before making a decision.

**Comment by aidanhs** [comment]
To be more explicit - it's not likely that the infra team will have a chance to proactively look into this.

As stated above, we would be looking for help on understanding:
 - how does this affect our CI times
 - how much of a 'significant bit' is this check (i.e. how often would this catch something that the current check wouldn't)

**Comment by pietroalbini** [comment]
Discussed this again in the infrastructure team meeting: we're still waiting for the data before making a decision, and unfortunately nobody in the team has a chance to gather it.

**Comment by RalfJung** [comment]
I am also not sure when I will have the time to do such experiments.

**Comment by RalfJung** [comment]
> @Mark-Simulacrum self-assigned this 2 days ago

Can I take this to mean you plan to look into this? :)

**Comment by Mark-Simulacrum** [comment]
Yes, I've been running out of time these past couple days but hopefully will get to it soon.

**Comment by Mark-Simulacrum** [comment]
As of #72053, we are running UI tests in check mode on the PR CI builder. We are not yet running them in 32 bit mode, though, because LLVM 8 is broken on cross-compilation (IIRC, don't remember details though).
